### ISSUE TITLE
Special case for single non-unity third-body efficiency 

### ISSUE THREAD

**Issue by olivecha** [AMReX-Combustion/PelePhysics#327]
Hi,

I stumbled upon some weird behaviour when using CEPTR to convert Cantera mechanism files to the format used in PeleLM. 

So, for example  in `Support/ceptr/utilities.py`, if the reaction has only one efficiency the third body is added to the reactants and products : 
```python
if hasattr(reaction, "efficiencies"):
        if len(reaction.efficiencies) == 1:
            reagents = copy.deepcopy(
                dict(
                    sum(
                        (
                            Counter(x)
                            for x in [reagents, reaction.efficiencies]
                        ),
                        Counter(),
                    )
                )
            )
```

This works well for reactions like reaction 35 in **grimech12** (H + O2 + H2O <=> HO2 + H2O) because H2O is added to the reactants and products.

But I am converting a mechanism where there is a third body reaction such as : 

NO2 + M <=> NO + O + M  (see Reaction 52 in [Jiang.yaml](https://github.com/AMReX-Combustion/PelePhysics/files/9816498/Jiang.txt))

With only H2O specified in the efficiences : 

`efficiencies: {H2O: 16.25}`

In this cases the reaction is re-defined as : 

NO2 + 16.25 H2O <=> NO + O + 16.25 H2O

Which alters the results when comparing PeleLM and Cantera data.

I feel like this is a special case where things break  because modifying CEPTR to handle the special case  : 

`len(reaction.efficienties) == 1 AND reaction.efficienties.values != 1.0 `

produces the same mechanism.H and mechanism.cpp files for all the mechanism shipped with PeleLM. 

I have put the special case handling in a fork : [olivecha/PelePhysics/ceptr-fix](https://github.com/olivecha/PelePhysics/tree/ceptr-fix)

with a [regression test](https://github.com/olivecha/PelePhysics/blob/ceptr-fix/Support/ceptr/tests/test_regression.py) to ensure changes do not affect the generation of the supplied mechanisms.

**Comment by marchdf** [comment]
Hi, thanks for bringing this to our attention. We will look into it but, in the mean time, would you mind creating a pull request with your branch? If it's the right fix, we can then easily merge it.

**Comment by malihass** [comment]
Thanks for reporting this.
Could you please let me know for what precise case you saw a difference between Cantera and PelePhysics? Using your mechanism, I was not able to see much of a difference on a 0D reactor, but maybe I did not try a test that actually highlighted the problem.

**Comment by olivecha** [comment]
Hi, 

I compared a Cantera 1D FreeFlame with a PeleLM case based on the FlameSheet example, but with a narrower domain and no perturbations. 

I used PMF data generated with the mechanism in Cantera. 

The flame speed, temperature, heat release, and species concentration at equilibrium converged to different values.

For example the NO2 mass fraction versus temperature  for PeleLM, PeleLMeX and Cantera : 

<img width="400" alt="image" src="https://user-images.githubusercontent.com/78630053/197095331-4252a260-a282-44d4-bc4f-f39c7a263c07.png">

I found the initial instantaneous reaction rates were different for certain species between PeleLM and Cantera. After modifying CEPTR I obtained a good concordance between PeleLMeX and Cantera for two mechanisms (Jiang supplied and SanDiego) :

<img width="400" alt="image" src="https://user-images.githubusercontent.com/78630053/197096095-55a50596-1d99-4bb1-8047-8cc92f1ca27b.png">
 
I can provide the PeleLM files used if needed.
